libraries/AP_Generator/AP_Generator_RichenPower.cpp

```diff
@@ -185,45 +225,67 @@ void AP_Generator_RichenPower::update_runstate()
     static const uint16_t SERVO_PWM_IDLE = 1500;
     static const uint16_t SERVO_PWM_RUN = 1900;
 
-    switch (runstate) {
+    // if the vehicle crashes then we assume the pilot wants to stop
+    // the motor.  This is done as a once-off when the crash is
+    // detected to allow the operator to rearm the vehicle, or we end
+    // up in a catch-22 situation where we force the stop state on the
+    // generator so they can't arm and can't start the generator
+    // because the vehicle is crashed.
+    if (AP::vehicle()->is_crashed()) {
+        if (!vehicle_was_crashed) {
+            gcs().send_text(MAV_SEVERITY_INFO, "Crash; stopping generator");
+            pilot_desired_runstate = RunState::STOP;
+            vehicle_was_crashed = true;
+        }
+    } else {
+        vehicle_was_crashed = false;
+    }
+
+    if (commanded_runstate != pilot_desired_runstate &&
+        !AP::arming().is_armed()) {
```

Comment:
Normally we use hal->soft_armed() to check if the vehicle is armed or not instead of using the AP_Arming library.  I don't care which one we use but I think using them inconsistently is dangerous.